uber/site_sections/budget.py

```diff
@@ -46,3 +46,157 @@ def mpoints(self, session):
         all = [(sum(mpu.amount for mpu in mpus), group, mpus)
                for group, mpus in groups.items()]
         return {'all': sorted(all, reverse=True)}
+
+    @ajax
+    def add_promo_code_words(self, session, text='', part_of_speech=0):
```

Comment:
The promo code words take the place of the PROMO_CODE_STUB config settings. I mentioned this in another comment, but I want to avoid adding anything to the config that could be done by an admin and saved in the database.

The up-front work we do now will save a huge amount of time down the road. I guarantee that if people start using promo codes, we'll get a never-ending stream of requests for new promo code words in the config. It's a huge development time sink that would be better spent writing code.

There's no reason why we should spend software development time modifying config settings and redeploying servers, when an admin could just as easily make the changes directly in the user interface.